stl/inc/regex

```diff
@@ -2382,6 +2382,7 @@ public:
     using pointer           = const value_type*;
     using reference         = const value_type&;
     using iterator_category = forward_iterator_tag;
+    using iterator_concept  = input_iterator_tag;

```

Comment:
It seems that we should only add `iterator_concept` if `_HAS_CXX20` (same for `regex_token_iterator`).